app/models/card.rb

```diff
@@ -20,15 +22,49 @@ def the_text_cannot_be_the_same
     end
   end
 
-  def shift_date
-    self.review_date = Date.today + 3.days
+  def set_review_date
+    self.review_date = DateTime.now
   end
 
   def check_translation(user_translation)
     if user_translation.mb_chars.downcase.strip == original_text.mb_chars.downcase.strip
-      update_attributes(review_date: Date.today + 3.days)
+      fail_count = 0
     else
-      return false
+      increment(:fail_count)
+    end
+
+    update_attributes(fail_count: calculate_fail_count,
+                      number_of_reviews: calculate_number_of_reviews,
+                      review_date: calculate_new_review_date,)
+
+    return !(self.fail_count > 0)
+  end
+
+  def calculate_number_of_reviews
+    case fail_count
+    when 0 then [number_of_reviews + 1, 4].min
+    when 3 then 0
+    else number_of_reviews
+    end
+  end
+
+  def calculate_new_review_date
```

Comment:
Assignment Branch Condition size for calculate_new_review_date is too high. [22.8/15]